public/content/developers/docs/design-and-ux/dex-design-best-practice/index.md

```diff
@@ -0,0 +1,217 @@
+
+The bottom line is that it probably doesn’t make a huge difference to usability. There are a few things to bear in mind, however, which might sway you one way or the other.
+
+It’s been mildly interesting to see the fashion change with time. Uniswap initially had the token on the left, but has since moved it to the right. Sushiswap also made this change during a design upgrade. Most, but not all, protocols have followed suit.
+
+Financial convention traditionally puts the currency symbol before the number, e.g. $50, €50, £50, but we *say* 50 dollars, 50 Euros, 50 pounds.
+
+To the general user - especially someone who reads left to right, top to bottom - token on the right probably feels more natural.
+
+![A UI with tokens on the left](./13.png)
+
+Putting the token on the left and all the numbers on the right looks pleasingly symmetrical, which is a plus, but there is another downside to this layout.
+
+The law of proximity states that items that are close together are perceived as related. Accordingly, we want to place related items next to each other. The token balance is directly related to the token itself, and will change whenever a new token is selected. It therefore makes slightly more sense for the token balance to be next to the token select button. It could be moved underneath the token, but that breaks the symmetry of the layout.
+
+Ultimately, there are pluses and minuses for both options, but it is interesting how the trend appears to be towards token on the right.
+
+# Button behavior {#button-behavior}
+
+Don’t have a separate button for Approve. Also don’t have a separate click for Approve. The user wants to Swap, so just say “swap” on the button and initiate the approval as the first step. A modal can show progress with a stepper, or a simple “tx 1 of 2 - approving” notification.
+
+![A UI with separate buttons for approve and swap](./14.png)
+
+![A UI with one button that says approve](./15.png)
+
+## Button as contextual help {#button-as-contextual-help}
+
+The button can do double duty as an alert!
+
+This is actually a fairly unusual design pattern outside of Web3, but has become standard within it. This is a good innovation as it saves space, and keeps attention focused.
+
+If the main action - SWAP - is unavailable due to an error, the reason why can be explained with the button, e.g.:
+
+- switch network
+- connect wallet
+- various errors
+
+The button can also be **mapped to the action** that needs to be performed. For example, if the user cannot swap because they are on the wrong network, the button should say “switch to Ethereum”, and when the user clicks on the button, it should switch the network to Ethereum. This speeds up the user flow significantly.
+
+![Key actions being initiated from the main CTA](./16.png)
+
+![Error message shown within the main CTA](./17.png)
+
+## Build your own with this figma file {#build-your-own-with-this-figma-file}
```

Comment:
Just wondering if we have a link for this Figma file referenced here

Reply:
yes we do, ill commit a change